src/lib/uTop_history.ml

```diff
@@ -33,15 +35,18 @@ let strip_colors s =
     with
     | Not_found -> None
   in
-  let rec find_color_escapes offset =
-    match find_escape offset with
-    | None -> [offset, len]
-    | Some esc_offset ->
-      try
-        let i = String.index_from s esc_offset 'm' in
-        (offset, esc_offset) :: find_color_escapes (i + 1)
-      with
-      | Not_found -> [offset, len]
+  let find_color_escapes offset =
+    let rec aux acc offset =
+      match find_escape offset with
+      | None -> (offset, len) :: acc
+      | Some esc_offset ->
+         try
+           let i = String.index_from s esc_offset 'm' in
+           aux ((offset, esc_offset) :: acc) (i + 1)
+         with
+         | Not_found -> (offset, len) :: acc
+    in
+    List.rev (aux [] offset)
```

Comment:
Note that we do a `List.map` on the result, so we might as well not do this `List.rev` and use `List.rev_map`